zerver/lib/narrow.py

```diff
@@ -908,7 +908,7 @@ def update_narrow_terms_containing_with_operator(
         try:
             message = access_message(maybe_user_profile, message_id)
         except JsonableError:
-            return narrow
+            raise BadNarrowOperatorError(_("Invalid 'with' operator"))
```

Comment:
In case of web public request I guess it would be enough to just remove the `with` operator since it would raise `MissingAuthenticationError` in `get_messages_backend` of `views/message_fetch`